hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/keyvalue/KeyValueContainer.java

```diff
@@ -649,22 +649,23 @@ public void importContainerData(InputStream input,
 
   public void importContainerData(KeyValueContainerData originalContainerData)
       throws IOException {
+    //Set all metadata and state BEFORE parsing and verifying checksum
+    containerData.setState(originalContainerData.getState());
     containerData
         .setContainerDBType(originalContainerData.getContainerDBType());
     containerData.setSchemaVersion(originalContainerData.getSchemaVersion());
 
+    // rewriting the yaml file with new checksum calculation
+    update(originalContainerData.getMetadata(), true);
+
     if (containerData.hasSchema(OzoneConsts.SCHEMA_V3)) {
       // load metadata from received dump files before we try to parse kv
       BlockUtils.loadKVContainerDataFromFiles(containerData, config);
     }
 
     //fill in memory stat counter (keycount, byte usage)
+    //This step performs checksum verification; all metadata must be set beforehand
     KeyValueContainerUtil.parseKVContainerData(containerData, config);
```

Comment:
@Gargi-jais11 we should move the checksum verification block out of the parseKVContainerData if that is the case and move the verify checksum at the end of this function(after line number 667 original code). We don't want to update the state of the container till the point the entire container data (i.e. data + metadata) is imported. Maybe make the verifyChecksum block in the function parseKVContainerData inside an if flag and not verify checksum till the end of this function.

Reply:
cc: @ptlrs

Reply:
@swamirishi , the container checksum calculation includes the content of chunkPath and metaPath. See #7930 for a task for remove chunkPath and metaPath from container checksum calculation.  So the container checksum will change after container is fully imported. Verify the checksum in the downloaded replica after original code L667 is just not what we want, and it will fail.